packages/react-avatar/src/components/Avatar/Avatar.types.ts

```diff
@@ -12,8 +12,12 @@ export interface AvatarProps extends ComponentPropsCompat, React.HTMLAttributes<
    */
   label?: ShorthandPropsCompat<React.HTMLAttributes<HTMLElement>>;
 
-  /** Icon displayed when there's no image. */
-  icon?: ShorthandPropsCompat<React.HTMLAttributes<HTMLElement>>;
+  /**
+   * Icon to be displayed when the avatar doesn't have an image or name (or if getInitials returns an empty string).
+   *
+   * @defaultvalue Person20Regular (the default icon's size depends on the Avatar's size)
+   */
+  icon?: React.ReactElement;
```

Comment:
This is a breaking change and I suggest to avoid it. Can we keep `icon` as a real slot?

- It creates inconsistencies in API (with other components, too) as other slots are actually slots
- It breaks compatibility with 3rd party icons (or even ours, but from other versions):

```tsx
import { Avatar } from '@fluentui/react-avatar'
import { AcceptIcon } from '@fluentui/react-northstar'

<Avatar icon={<AcceptIcon />} />
//            👆 will it work?
```

Reply:
I've updated `icon` to be a real slot that is only rendered when there is no label (and updated some styling to make that work).

For what it's worth, your code snippet would have worked fine to render a 3rd party icon. The icon was being used as the label slot's children when there was no other label text. That's also how it was working even before this change--this PR was updating the API to match what the implementation was doing. But having it be a fully-fledged slot is better. Thanks for the feedback!